docs/fsharp-cheatsheet.md

```diff
@@ -555,136 +555,158 @@ Another way of implementing interfaces is to use *object expressions*.
 <a name="CodeOrganization_Modules"></a>
 ### Modules
 Modules are key building blocks for grouping related code; they can contain `types`, `let` bindings, or (nested) sub `module`s.
-Identifiers within modules can be referenced using dot notation, or you can bring them into scope via the `open` keyword. Illustrative-only example:
-
-    module Game =
-        let mutable basePoints = 1
-        type Player = { id: int; score: int }
-        let playerScored player = { player with score = player.score + basePoints }
-
-    let player: Game.Player = { id = 1; score = 0 }
-    let player' = Game.playerScored player  // score = 1
-
-    open Game
-    basePoints <- 2
-    let player'' = playerScored player'  // player''.score = 3
-
-If you have only one module in a file, the `module` name can be declared at the top of the file, with all further declarations
-being module elements (and non indentation required)
+Identifiers within modules can be referenced using dot notation, or you can bring them into scope via the [`open`](#CodeOrganization_OpenAndAutoOpen) keyword.
+Illustrative-only example:
+
+    module Money =
+        type CardInfo =
+            { number: string
+              expiration: int * int }
+
+        type Payment =
+            | Card of CardInfo
+            | Cash of int
+
+        module Functions =
+            let getCreditCardType (cardNumber: string) =
+                match cardNumber.ToCharArray() |> Array.truncate 2 with
```

Comment:
this is a good example to put somehere but its too long and a bit of a non sequiter here for explaining namespacing/opens. Also can trucate throw?

Reply:
Agreed. I'll shorten it up.

truncate: I'll test, but I chose it over take, which does throw.

Reply:
I can never remember truncate vs take but you are correct. Examples is still long for exmplaining namespaces tho!